src/js/components/Carousel/__tests__/Carousel-test.tsx

```diff
@@ -1,139 +1,157 @@
 import React from 'react';
-import { render, fireEvent, act } from '@testing-library/react';
+import { render, act, waitFor, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
 import 'jest-styled-components';
+import '@testing-library/jest-dom';
 
-import { Grommet } from '../../Grommet';
 import { Carousel } from '..';
-import { Image } from '../../Image';
+
+const getSlideOne = () => screen.getByText('Slide One');
+const getSlideTwo = () => screen.getByText('Slide Two');
+const getSlideThree = () => screen.getByText('Slide Three');
 
 describe('Carousel', () => {
   test('basic', () => {
     const { container } = render(
-      <Grommet>
-        <Carousel>
-          <Image src="//v2.grommet.io/assets/IMG_4245.jpg" />
-          <Image src="//v2.grommet.io/assets/IMG_4210.jpg" />
-        </Carousel>
-      </Grommet>,
+      <Carousel>
+        <div>Slide One</div>
+        <div>Slide Two</div>
+      </Carousel>,
     );
-
+    expect(getSlideOne()).toBeVisible();
+    expect(getSlideTwo()).not.toBeVisible();
     expect(container.firstChild).toMatchSnapshot();
   });
 
   test('basic with `initialChild: 1`', () => {
-    const { container } = render(
-      <Grommet>
-        <Carousel initialChild={1}>
-          <Image src="//v2.grommet.io/assets/IMG_4245.jpg" />
-          <Image src="//v2.grommet.io/assets/IMG_4210.jpg" />
-        </Carousel>
-      </Grommet>,
+    render(
+      <Carousel initialChild={1}>
+        <div>Slide One</div>
+        <div>Slide Two</div>
+      </Carousel>,
     );
 
-    expect(container.firstChild).toMatchSnapshot();
+    expect(getSlideTwo()).toBeVisible();
+    expect(getSlideOne()).not.toBeVisible();
   });
 
-  test('navigate', () => {
-    const { getByTestId, container } = render(
-      <Grommet>
-        <Carousel data-testid="test-carousel">
-          <Image src="//v2.grommet.io/assets/IMG_4245.jpg" />
-          <Image src="//v2.grommet.io/assets/IMG_4210.jpg" />
-        </Carousel>
-      </Grommet>,
+  test('arrow navigation: next', async () => {
+    render(
+      <Carousel controls="arrows">
+        <div>Slide One</div>
+        <div>Slide Two</div>
+        <div>Slide Three</div>
+      </Carousel>,
     );
-    expect(container.firstChild).toMatchSnapshot();
-
-    fireEvent.keyDown(getByTestId('test-carousel'), {
```

Comment:
I think we should have test coverage for keyboard interactions, right and left arrow keys.